apollo-router/src/spec/selection.rs

```diff
@@ -224,42 +234,56 @@ impl Selection {
                 if fragment.as_str() == type_condition.as_str() {
                     selection_set
                         .iter()
-                        .any(|selection| selection.contains_error_path(&path[1..], fragments))
+                        .map(|selection| {
+                            selection.matching_error_path_length(&path[1..], fragments)
+                        })
+                        .max()
+                        .unwrap_or(0)
+                        + 1
                 } else {
-                    false
+                    0
                 }
             }
             (Some(PathElement::Fragment(fragment)), Self::FragmentSpread { name, .. }) => {
                 if let Some(f) = fragments.get(name) {
                     if fragment.as_str() == f.type_condition.as_str() {
                         f.selection_set
                             .iter()
-                            .any(|selection| selection.contains_error_path(&path[1..], fragments))
+                            .map(|selection| {
+                                selection.matching_error_path_length(&path[1..], fragments)
+                            })
+                            .max()
+                            .unwrap_or(0)
+                            + 1
                     } else {
-                        false
+                        0
                     }
                 } else {
-                    false
+                    0
                 }
             }
             (_, Self::FragmentSpread { name, .. }) => {
                 if let Some(f) = fragments.get(name) {
                     f.selection_set
                         .iter()
-                        .any(|selection| selection.contains_error_path(path, fragments))
+                        .map(|selection| selection.matching_error_path_length(path, fragments))
+                        .max()
+                        .unwrap_or(0)
                 } else {
-                    false
+                    0
                 }
             }
             (Some(PathElement::Index(_)), _) | (Some(PathElement::Flatten(_)), _) => {
-                self.contains_error_path(&path[1..], fragments)
+                self.matching_error_path_length(&path[1..], fragments) + 1
```

Comment:
`&path[1..]` are we sure this call cannot panic ? what if we have a path size of 1 ? Does it panic ? I can't remember

Reply:
if path has 1 element, then `&path[1..]` is the empty slice.

We could use `path.split_first()` at the top of this match statement instead of `path.first()` and `&path[1..]` to compile time guarantee that there can't be a panic, but it should be fine already

Reply:
@IvanGoncharov I saw you used `split_first` but you're not using the second part of this `split_first`here instead of `&path[1..]`